charts/kargo/values.yaml

```
@@ -106,6 +106,9 @@ api:
   ## @param api.podAnnotations Optional annotations to add to pods. Merges with `global.podAnnotations`, allowing you to override or add to the global annotations.
   podAnnotations: {}
 
+  ## @param api.enableSecretManagement Specifies whether secret management is enabled. When enabled, the API server will manage secrets as allowed by its configured permissions. If disabled, the API server will not manage secrets, and related operations will be restricted or result in unimplemented responses. This setting enhances security by limiting the API server's permissions and reducing its attack surface. If secret management is not needed, it is advisable to disable this feature.
```

I propose rewording a little bit so that it is more clear that this is a choice one might make if they are GitOps'ing their Kargo Projects -- which is a common thing in larger, more security-conscious organizations.

```suggestion
  ## @param api.enableSecretManagement Specifies whether Secret management is enabled. This affects the API server's ability to manage repository credentials and other Project-level Secrets, such as those used by AnalysisRuns for verification purposes. If using GitOps to manage Kargo Projects declaratively, the API's Secret management capabilities are not needed and can be disabled to effectively reduce the API server's attackable surface.
```

Note that codegen needs to be re-run after this.